chroot-script

```diff
@@ -616,19 +616,31 @@ grub_install() {
      for device in $SELECTED_PARTITIONS ; do
         GRUB="${device%%[0-9]}"
         echo "Installing grub on ${GRUB}:"
-        grub-install --no-floppy "$GRUB"
+        if ! grub-install --no-floppy "$GRUB" ; then
+          echo "Error: failed to execute 'grub-install --no-floppy $GRUB'." >&2
+          exit 1
+        fi
+
      done
      rm -f /boot/grub/device.map
   else
      echo "Installing grub on ${GRUB}:"
      case "$RELEASE" in
        lenny|squeeze|wheezy)
-         grub-install --no-floppy "$(readlink -f "${GRUB}")"
+         local grub_dev
+         grub_dev="$(readlink -f "${GRUB}")"
```

Comment:
not sure about the quoting here?

Reply:
You mean because of the `"${GRUB}"` inside `"$(...)"`. If so, that's expected as such, demonstration:

```
% cat foo 
#!/bin/bash
DEV="/dev/foo bar"
foo "$(readlink -f ${DEV})"
% shellcheck foo

In foo line 3:
foo "$(readlink -f ${DEV})"
                   ^-- SC2086: Double quote to prevent globbing and word splitting.
```